### PR TITLE
Prepend queue ids

### LINKED ISSUE #1245: Add ids to front of user queue
Adding subjects to `subject_queue` should allow the option of adding to the front of the queue for use in projects like spacewarps retirment scheme.

### PR DESCRIPTION
**Pull request by camallen**
closes #1245 - does what it says on the tin.

I'd rather have a `has_many :subjects, through: :set_member_subjects` relation on SubjectQueue working but the has_many_through scope didn't work with the belongs_to_many link.

### CODE REVIEW

**Comment by edpaget** [comment]
LGtM

**Comment by camallen** [comment]
i've got to run rspec bisect on this to get a reproducable failing case.

File: spec/controllers/api/v1/subject_queues_controller_spec.rb [hunk 1]
```diff
@@ -31,18 +34,50 @@
 
   describe "#update" do
     let(:test_attr) { :set_member_subject_ids }
-    let(:test_attr_value) { subjects.map(&:id) }
+    let(:test_attr_value) { subject_ids }
```
Comment (edpaget) [inline spec/controllers/api/v1/subject_queues_controller_spec.rb]: These should be sms_ids

File: spec/controllers/api/v1/subject_queues_controller_spec.rb [hunk 2]
```diff
@@ -31,18 +34,50 @@
...
+    let(:test_relation_ids) { subject_ids }
```
Comment (edpaget) [inline spec/controllers/api/v1/subject_queues_controller_spec.rb]: So should this.

Comment (camallen) [inline spec/controllers/api/v1/subject_queues_controller_spec.rb]: the api requires subject ids to be passed in which are then converted to sms ids on the fly, so i need different expectations. I'm running bisect to get a reproducible test case on this.

**Pull request merged**